src/ucp/api/ucp.h

```diff
@@ -1293,6 +1293,31 @@ ucs_status_t ucp_mem_map(ucp_context_h context, ucp_mem_map_params_t *params,
 ucs_status_t ucp_mem_unmap(ucp_context_h context, ucp_mem_h memh);
 
 
+enum ucp_mem_advice_flags {
+    UCP_MADV_NORMAL     = 0,         /* No special treatment */
+    UCP_MADV_WILLNEED   = UCS_BIT(0) /* Expect access in the near future */ 
+};
+/**
+ * @ingroup UCP_MEM
+ * @brief give advice about use of memory
+ *
+ * Advises the UCP about how to handle memory range beginning at address and
+ * size of length bytes. This call does not influence the semantics of the 
+ * application, but may influence its performance. The UCP is free to ignore
+ * the advice
```

Comment:
Please add some examples (not code but logical description) when it can be useful and how user can leverage this routine.